.github/workflows/release.yml

```diff
@@ -29,22 +35,27 @@ jobs:
       uses: actions/checkout@v4
 
     - uses: dtolnay/rust-toolchain@stable
+      with:
+        targets: ${{ matrix.target }}
+
+    - uses: taiki-e/install-action@cargo-zigbuild
+    - uses: mlugg/setup-zig@v2
 
     - name: Extract release notes
       shell: bash
       run: |
         awk "/# ${GITHUB_REF_NAME}/{flag=1; next} /^$/{flag=0} flag" CHANGELOG.md >REL.md
 
     - name: Build stage1
-      run: cargo build --release
+      run: cargo zigbuild --release --target ${{ matrix.target }}
 
     - name: Switch to linking with wild
       run: |
-        mkdir .cargo
-        echo -e "[target.${{matrix.target}}]\nlinker = \"clang\"\nrustflags = [\"-Clink-arg=--ld-path=${PWD}/target/release/wild\"]\n" >>"$HOME/.cargo/config.toml"
+        mkdir -p "$HOME/.cargo"
+        echo -e "[target.${{matrix.target}}]\nlinker = \"clang\"\nrustflags = [\"-Clink-arg=--ld-path=${PWD}/target/${{matrix.target}}/release/wild\"]\n" >>"$HOME/.cargo/config.toml"
 
     - name: Build release binary
-      run: cargo build --profile dist --target ${{ matrix.target }}
+      run: cargo zigbuild --profile dist --target ${{ matrix.target }}
```

Comment:
The musl allocator is pretty slow. I'd suggest adding `--features mimalloc`, but only for the musl builds. I just benchmarked and a musl build with mimalloc was about 20% faster than a musl build without.